api/src/main/java/com/nicknathanjustin/streamercontracts/donations/DonationsApiController.java

```diff
@@ -19,12 +25,30 @@
 
     @NonNull private final PaymentsService paymentsService;
 
+    @NonNull private final ContractService contractService;
+
+    @NonNull private final UserService userService;
+
+    @NonNull private final DonationService donationService;
+
     @RequestMapping(path = "/execute", method = RequestMethod.POST)
     public String executePayment(@RequestBody @NonNull final CreateDonationRequest createDonationRequest) {
         final Payment payment = paymentsService.executePayment(createDonationRequest.getPayPalPaymentId()).orElse(null);
         if(payment == null) {
             return "couldn't execute payment for id: " + createDonationRequest.getPayPalPaymentId();
         }
+
+        final String streamerName = "CrazyHorse";
+        final Optional<UserModel> proposerObj = userService.getUser(createDonationRequest.getUsername());
+        final Optional<UserModel> streamerObj = userService.getUser(streamerName);
+        if (!proposerObj.isPresent() || !streamerObj.isPresent()) {
+            return "users do not exist";
```

Comment:
This should return HTTP.InvalidRequest or something like that i believe. Additionally this feels like a good place to log something like ->

```
log.warn(Attempted to create donation for non existant user: + userNameThatRetunedNull)
```